### PR TITLE
Implement geofencing settings in the project properties' QField panel

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds a couple of widgets to setup geofencing in QField. Looks:

image

The libqfieldsync counterpart is here: https://github.com/opengisch/libqfieldsync/pull/86

### CODE REVIEW

**Comment by m-kuhn** [comment]
Nice !

Can we have a UI adjustment: the description what it does (i.e. it will alert in case of XXX) should be there. The "reverted logic" should live as a tooltip instead.

**Comment by nirvn** [comment]
@m-kuhn , we can certainly have a UI adjustment ;) good idea.

File: qfieldsync/gui/project_configuration_widget.py [hunk 1]
```diff
@@ -229,21 +269,56 @@ def apply(self):
...
             )
         except AttributeError:
             pass
+
+        # Geofencing settings
+        self.__project_configuration.geofencing_is_active = (
+            self.geofencingGroupBox.isChecked()
+        )
+
+        try:
```
Comment (suricactus) [inline qfieldsync/gui/project_configuration_widget.py:304]: Do you have a concrete example when we need the `try/except`? I see it is also 10 lines above, but on this line it seems you handle this with the `if/else` check for `currentLayer()` existance.

**Review by suricactus — APPROVED** [review]
Looks good to me, one small comment.

**Pull request merged**